applications/nrf_desktop/src/modules/usb_state.c

```diff
@@ -138,7 +140,7 @@ static int get_report(const struct device *dev, struct usb_setup_packet *setup,
 	return -ENOTSUP;
 }
 
-static int set_report(const struct device *dev, struct usb_setup_packet *setup, int32_t *len, uint8_t **data)
+static int set_report(const struct device *dev, struct usb_setup_packet *setup, int32_t *len_to_ignore, uint8_t **data)
```

Comment:
Argument passed by len is still valid (is equal to setup->wLength), please consider changing it to:
```suggestion
static int set_report(const struct device *dev,
                     struct usb_setup_packet *setup,
                     const int32_t *len, uint8_t **data)
```

Reply:
Why did you add const here for the third argument? This would not work!

Reply:
I do not see why it should not work.

Reply:
```
[2021-11-16T14:09:05.761Z] FAILED: CMakeFiles/app.dir/src/modules/usb_state.c.obj 

[2021-11-16T14:09:05.761Z] ccache /opt/gnuarmemb/gcc-arm-none-eabi-9-2019-q4-major/bin/arm-none-eabi-gcc -DBUILD_VERSION=v2.6.99-ncs1-2245-g35654ffa6d06 -DKERNEL -DNRF52820_XXAA -DUSE_PARTITION_MANAGER=0 -D_FORTIFY_SOURCE=2 -D__PROGRAM_START -D__ZEPHYR__=1 -I/jenkins_workspace/workspace/latest_sdk-nrf_PR-6079/nrf/applications/nrf_desktop/src/events -I/jenkins_workspace/workspace/latest_sdk-nrf_PR-6079/nrf/applications/nrf_desktop/src/util -I/jenkins_workspace/workspace/latest_sdk-nrf_PR-6079/zephyr/include -Izephyr/include/generated -I/jenkins_workspace/workspace/latest_sdk-nrf_PR-6079/zephyr/soc/arm/nordic_nrf/nrf52 -I/jenkins_workspace/workspace/latest_sdk-nrf_PR-6079/zephyr/subsys/bluetooth -I/jenkins_workspace/workspace/latest_sdk-nrf_PR-6079/zephyr/subsys/usb -I/jenkins_workspace/workspace/latest_sdk-nrf_PR-6079/zephyr/subsys/settings/include -I/jenkins_workspace/workspace/latest_sdk-nrf_PR-6079/nrf/include -I/jenkins_workspace/workspace/latest_sdk-nrf_PR-6079/nrf/subsys/event_manager/. -I/jenkins_workspace/workspace/latest_sdk-nrf_PR-6079/nrf/subsys/event_manager_profiler/. -I/jenkins_workspace/workspace/latest_sdk-nrf_PR-6079/modules/hal/cmsis/CMSIS/Core/Include -I/jenkins_workspace/workspace/latest_sdk-nrf_PR-6079/modules/hal/nordic/nrfx -I/jenkins_workspace/workspace/latest_sdk-nrf_PR-6079/modules/hal/nordic/nrfx/drivers/include -I/jenkins_workspace/workspace/latest_sdk-nrf_PR-6079/modules/hal/nordic/nrfx/mdk -I/jenkins_workspace/workspace/latest_sdk-nrf_PR-6079/zephyr/modules/hal_nordic/nrfx/. -I/jenkins_workspace/workspace/latest_sdk-nrf_PR-6079/modules/crypto/tinycrypt/lib/include -I/jenkins_workspace/workspace/latest_sdk-nrf_PR-6079/nrf/applications/nrf_desktop/configuration/common -I/jenkins_workspace/workspace/latest_sdk-nrf_PR-6079/nrf/applications/nrf_desktop/configuration/nrf52820dongle_nrf52820 -I/jenkins_workspace/workspace/latest_sdk-nrf_PR-6079/nrfxlib/softdevice_controller/include -isystem /jenkins_workspace/workspace/latest_sdk-nrf_PR-6079/zephyr/lib/libc/minimal/include -isystem /opt/gnuarmemb/gcc-arm-none-eabi-9-2019-q4-major/bin/../lib/gcc/arm-none-eabi/9.2.1/include -isystem /opt/gnuarmemb/gcc-arm-none-eabi-9-2019-q4-major/bin/../lib/gcc/arm-none-eabi/9.2.1/include-fixed -O2 -imacros /jenkins_workspace/workspace/latest_sdk-nrf_PR-6079/twister-build/nrf52820dongle_nrf52820/nrf_desktop/applications.nrf_desktop.zdebug/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -gdwarf-4 -fdiagnostics-color=always -mcpu=cortex-m4 -mthumb -mabi=aapcs -mfp16-format=ieee -imacros /jenkins_workspace/workspace/latest_sdk-nrf_PR-6079/zephyr/include/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-address-of-packed-member -Wno-unused-but-set-variable -Werror=implicit-int -Werror -fno-asynchronous-unwind-tables -fno-pie -fno-pic -fno-strict-overflow -fno-reorder-functions -fno-defer-pop -fmacro-prefix-map=/jenkins_workspace/workspace/latest_sdk-nrf_PR-6079/nrf/applications/nrf_desktop=CMAKE_SOURCE_DIR -fmacro-prefix-map=/jenkins_workspace/workspace/latest_sdk-nrf_PR-6079/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/jenkins_workspace/workspace/latest_sdk-nrf_PR-6079=WEST_TOPDIR -ffunction-sections -fdata-sections -std=c99 -nostdinc -MD -MT CMakeFiles/app.dir/src/modules/usb_state.c.obj -MF CMakeFiles/app.dir/src/modules/usb_state.c.obj.d -o CMakeFiles/app.dir/src/modules/usb_state.c.obj -c /jenkins_workspace/workspace/latest_sdk-nrf_PR-6079/nrf/applications/nrf_desktop/src/modules/usb_state.c

[2021-11-16T14:09:05.761Z] /jenkins_workspace/workspace/latest_sdk-nrf_PR-6079/nrf/applications/nrf_desktop/src/modules/usb_state.c: In function 'usb_init':

[2021-11-16T14:09:05.761Z] /jenkins_workspace/workspace/latest_sdk-nrf_PR-6079/nrf/applications/nrf_desktop/src/modules/usb_state.c:634:17: error: initialization of 'int (*)(const struct device *, struct usb_setup_packet *, int32_t *, uint8_t **)' {aka 'int (*)(const struct device *, struct usb_setup_packet *, int *, unsigned char **)'} from incompatible pointer type 'int (*)(const struct device *, struct usb_setup_packet *, const int32_t *, uint8_t **)' {aka 'int (*)(const struct device *, struct usb_setup_packet *, const int *, unsigned char **)'} [-Werror=incompatible-pointer-types]

[2021-11-16T14:09:05.761Z]   634 |   .set_report = set_report,

[2021-11-16T14:09:05.761Z]       |                 ^~~~~~~~~~
```

Reply:
sorry, there is typedef, but I can change that upstream later.